src/Functional.CQS.AOP.CommonTestInfrastructure.MetricsCapturing/MetricsCapturingTestUtility.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using Functional.CQS.AOP.CommonTestInfrastructure.DummyObjects;
+using Functional.CQS.AOP.IoC.PureDI.MetricsCapturing;
+
+namespace Functional.CQS.AOP.CommonTestInfrastructure.MetricsCapturing
+{
+	/// <summary>
+	/// Contains common test infrastructure for metrics.
+	/// </summary>
+	public static class MetricsCapturingTestUtility
+	{
+		/// <summary>
+		/// Gets the collection of Functional.CQS metrics-capturing strategy implementation types associated with elements in <see cref="TestUtility.CQSHandlerContractTypes"/>.
+		/// </summary>
+		public static IReadOnlyDictionary<Type, Type> MetricsCapturingDecoratorTypeLookupByCQSHandlerContractType { get; } = new Dictionary<Type, Type>()
+		{
+			{ typeof(IQueryHandler<DummyQueryReturnsValueType, DummyQueryReturnsValueTypeResult>), typeof(QueryHandlerMetricsCapturingDecorator<DummyQueryReturnsValueType, DummyQueryReturnsValueTypeResult>) },
+			{ typeof(IQueryHandler<DummyQueryReturnsReferenceType, DummyQueryReturnsReferenceTypeResult>), typeof(QueryHandlerMetricsCapturingDecorator<DummyQueryReturnsReferenceType, DummyQueryReturnsReferenceTypeResult>) },
+			{ typeof(IAsyncQueryHandler<DummyAsyncQueryReturnsValueType, DummyAsyncQueryReturnsValueTypeResult>), typeof(AsyncQueryHandlerMetricsCapturingDecorator<DummyAsyncQueryReturnsValueType, DummyAsyncQueryReturnsValueTypeResult>) },
+			{ typeof(IAsyncQueryHandler<DummyAsyncQueryReturnsReferenceType, DummyAsyncQueryReturnsReferenceTypeResult>), typeof(AsyncQueryHandlerMetricsCapturingDecorator<DummyAsyncQueryReturnsReferenceType, DummyAsyncQueryReturnsReferenceTypeResult>) },
+			{ typeof(ICommandHandler<DummyCommandThatSucceeds, DummyCommandError>), typeof(CommandHandlerMetricsCapturingDecorator<DummyCommandThatSucceeds, DummyCommandError>) },
+			{ typeof(ICommandHandler<DummyCommandThatFails, DummyCommandError>), typeof(CommandHandlerMetricsCapturingDecorator<DummyCommandThatFails, DummyCommandError>) },
+			{ typeof(IAsyncCommandHandler<DummyAsyncCommandThatSucceeds, DummyAsyncCommandError>), typeof(AsyncCommandHandlerMetricsCapturingDecorator<DummyAsyncCommandThatSucceeds, DummyAsyncCommandError>) },
+			{ typeof(IAsyncCommandHandler<DummyAsyncCommandThatFails, DummyAsyncCommandError>), typeof(AsyncCommandHandlerMetricsCapturingDecorator<DummyAsyncCommandThatFails, DummyAsyncCommandError>) }
+		};
+
+		/// <summary>
+		/// Gets the collection of Functional.CQS metrics-capturing strategy implementation types associated with elements in <see cref="TestUtility.CQSHandlerContractTypes"/>.
```

Comment:
The comment for the Universal decorator lookup is identical to the comment for the non-Universal lookup.
The comments may need to distinguish Universal from non-Universal?

Reply:
Fixed